### ISSUE TITLE
exec: make tracing of cfetcher similar to row fetcher

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#38807]
At the moment, cfetcher emits less messages about its operations into the trace. Here is an example of logic test failure when vectorize is set to streaming:
```
        --- FAIL: TestPlannerLogic/local/update (0.36s)
            logic.go:2365: 
                 
                testdata/planner_test/update:18: SELECT message FROM [SHOW KV TRACE FOR SESSION]
                 WHERE message LIKE 'fetched:%' OR message LIKE 'output row%'
                expected:
                    fetched: /kv2/primary/'a' -> NULL
                    fetched: /kv2/primary/'a'/v -> 'b'
                    output row: ['a' 'b']
                    fetched: /kv2/primary/'c' -> NULL
                    fetched: /kv2/primary/'c'/v -> 'd'
                    output row: ['c' 'd']
                    fetched: /kv2/primary/'e' -> NULL
                    fetched: /kv2/primary/'e'/v -> 'f'
                    output row: ['e' 'f']
                    fetched: /kv2/primary/'f' -> NULL
                    fetched: /kv2/primary/'f'/v -> 'g'
                    output row: ['f' 'g']
                    
                but found (query options: "") :
                    output row: ['a' 'b']
                    output row: ['c' 'd']
                    output row: ['e' 'f']
                    output row: ['f' 'g']
```
The following logic tests are failing (note you can add `local-vec` config to the file to reproduce):
```
TestPlannerLogic/local/update
TestPlannerLogic/local/aggregate
TestPlannerLogic/local/ddl
TestPlannerLogic/local/select
TestPlannerLogic/local/select_index
TestPlannerLogic/local/insert
TestPlannerLogic/local/interleaved
TestPlannerLogic/local/order_by
```
All of the above are missing information about `fetched` row.

This one (`TestLogic/local/select_index`) is missing `Scan /Table/65/1/1{-/#}` as well as a bunch of other things (but the test requires only "Scan ...").

I spent some time trying to make it work to no progress. Assigning to Jordan for now.

**Comment by rohany** [comment]
Isn't the output going to be a bit different anyway once we turn on streaming? Because we read in an entire batch of elements at once, the KV trace is going to be different than reading in one k/v pair at a time (at least my understanding).

**Comment by yuzefovich** [comment]
I think a different output is ok. The goal here to provide as much information in tracing as possible. If it doesn't match exactly what row fetcher does, then we'll adjust the tests.

**Comment by rohany** [comment]
It seems strange to change the test though -- does that mean we don't want to be verifying distsql's behavior anymore?

**Comment by yuzefovich** [comment]
Hm, maybe we'll have different logic test files with different configs then, I'm not sure. But a different output sounds good to me as long as it is as informative as possible. Probably Jordan has more thoughts on this.

**Comment by jordanlewis** [comment]
No, the output does not have have to be at all different from what it was before. The fetcher is the point at which we convert kv-by-kv data (what it looks like on disk) into column vector batches. The fetcher still reads everything in the same order as it used to.

**Comment by yuzefovich** [comment]
Closing this issue since Rohan's work has addressed all the differences.

**Comment by yuzefovich** [comment]
I closed it a little prematurely - this will be fixed once #38945 is merged.
